components/ratgdo/ratgdo.cpp

```diff
@@ -603,10 +611,25 @@ namespace ratgdo {
     {
         data |= (1 << 16); // button 1 ?
         data |= (1 << 8); // button press
-        this->send_command(Command::DOOR_ACTION, data, false);
-        set_timeout(200, [=] {
-            auto data2 = data & ~(1 << 8); // button release
-            this->send_command(Command::DOOR_ACTION, data2);
+        this->send_command(Command::DOOR_ACTION, data, false, [=]() {
+            set_timeout(100, [=] {
+                auto data2 = data & ~(1 << 8); // button release
+                this->send_command(Command::DOOR_ACTION, data2);
+            });
+        });
+    }
+
+    void RATGDOComponent::ensure_door_command(uint32_t data, uint32_t delay)
+    {        
+        this->door_state_received.then([=](DoorState s) {
+            // TODO: check door state is the expected one for command data?
```

Comment:
Is there still more to do here or is this ready for review?

Reply:
The TODO comment was there because I was undecided if further checking of the door state after was warranted... I ended up adding it. Should be ready for review now.